web/src/date-utils.ts

```diff
@@ -54,31 +33,46 @@ export const dateToFormat = (date: Date | number, format: DateFormat): string =>
   if (isDateValid(date)) {
     switch (format) {
       case DateFormat.TimeShort:
-        return timeShortFormatter.format(date);
+        {
+          const dateObject = new Date(date);
```

Comment:
Nit: Can move `const dateObject = new Date(date);` outside the `switch` since all `case` branches use it and we already checked that it is a valid date.